tests/test_generation_utils.py

```diff
@@ -1903,3 +1903,122 @@ def test_generate_encoder_outputs_attention_mask(self):
         output_sequences_with_mask = output_sequences_with_mask.cpu()
 
         self.assertListEqual(output_sequences_no_mask.tolist(), output_sequences_with_mask.tolist())
+
+    @staticmethod
+    def compute_transition_beam_scores_sum(outputs, vocab_size):
+        # reshape scores
+        scores = torch.stack(outputs.scores).reshape(len(outputs.scores), -1).transpose(0, 1)
+        cut_idx = outputs.sequences.shape[-1] - scores.shape[-1]
+        # compute real indices
+        indices = (
+            outputs.sequences[:, cut_idx:]
+            + torch.tensor(outputs.beam_indices, device=outputs.sequences.device) * vocab_size
+        )
+
+        # gather scores and run
+        return scores.gather(0, indices).sum(-1)
+
+    def test_transition_scores_beam_search_encoder_decoder(self):
+        articles = [
+            "Justin Timberlake and Jessica Biel, welcome to parenthood.",
+            "Michael Phelps is arguably the most decorated Olympian of all time.",
+        ]
+        tokenizer = BartTokenizer.from_pretrained("hf-internal-testing/tiny-random-bart")
+        model = BartForConditionalGeneration.from_pretrained(
+            "hf-internal-testing/tiny-random-bart",
+            max_length=10,
+            num_beams=4,
+            num_return_sequences=2,
+            eos_token_id=None,
+            return_dict_in_generate=True,
+            output_scores=True,
+            length_penalty=0.0,
+        )
+        model = model.to(torch_device)
+
+        input_ids = tokenizer(articles, return_tensors="pt", padding=True).input_ids.to(torch_device)
+        outputs = model.generate(input_ids=input_ids)
+
+        transition_scores_sum = self.compute_transition_beam_scores_sum(outputs, model.config.vocab_size)
+
+        self.assertTrue(torch.allclose(transition_scores_sum, outputs.sequences_scores, atol=1e-3))
```

Comment:
Very important test, I'm glad it was included. The workaround I posted in the original issue did not give the exact same scores as `sequences_scores`, so it's good that this approach does.